### DISCUSSION TITLE
NextAuth custom session and AuthenticatedItem : user not logged in in UI

### DISCUSSION THREAD

**Discussion by marionLa** [keystonejs/keystone#9061] [Questions]
Hi, 
I authenticate by NextAuth in my application by following [this example](https://github.com/keystonejs/keystone/tree/main/examples/custom-session-next-auth )  The login/callback process goes well but the user is not displayed on the dashboard, and the authenticatedItem returns '{state: 'unauthenticated'} I've gone through the threads (e.g. #8919) but haven't found anything that solves my problem. Does anyone have an idea?

**Comment by marekryb** [comment]
Hi,

`authenticatedItem` is a query implemented by `@keystone-6/auth` package, which (in most cases) you should **NOT** be using together with next-auth.
Beside that it is hard to know what is the problem, unless you share some code.

**Reply by marionLa** [reply]
Sorry for the late response... Ok i understand for the auth package. My code is exactly the same as [the example ](https://github.com/keystonejs/keystone/tree/main/examples/custom-session-next-auth), except the sessionSecret, the github id and the github secret. I have deleted auth.ts and replace it by session.ts, and modified the keystone.ts like the example. 
My homepage looks like the screenshot below
![Capture d’écran du 2024-03-19 16-00-48](https://github.com/keystonejs/keystone/assets/81257309/cd105611-0a75-4c49-9727-21558bae0547)
Instead of having the logged user in the sidenav, i have the "standard menu". 
I have three cookies in the storage : next-auth-callback-url, next-auth-csrf token, and next-auth-session-token.
Maybe it's not possible to show the logged in user informations  by this way ? Or did i miss something ? 
Thanks for your help !

**Reply by marekryb** [reply]
Ok, it now clicked for me that you just have a problem with admin UI.
You can check the navigation panel code here: https://github.com/keystonejs/keystone/blob/041dcf7c2ab9950da940a35486a4c5570462840a/packages/core/src/admin-ui/components/Navigation.tsx#L63
The dialog with username is not showing because `authenticatedItem` query does not exist. Which leaves you at least two options:
1) Implement your own [navigation panel](https://keystonejs.com/docs/guides/custom-admin-ui-navigation) and instead of using `authenticatedItem` prop (will be always empty), you would have to use `next-auth`'s client API like [useSession()](https://next-auth.js.org/getting-started/client#usesession). Then you can either pass session info to standard `NavigationContainer` (it expect specific format from `@keystone-6/auth`) or you can just create your own.
2) (untested) [Create a custom query](https://keystonejs.com/docs/guides/schema-extension) called `authenticatedItem` that would use server-side `next-auth` API ([getServerSession()](https://next-auth.js.org/configuration/nextjs#getserversession)) and returns data in same format as `@keystone-6/auth` does.
